app/models/device_group.rb

```diff
@@ -0,0 +1,7 @@
+class DeviceGroup < ApplicationRecord
+  has_many :devices, dependent: :destroy
+  belongs_to :organization, class_name: 'Organization', foreign_key: 'organization_id', optional: true
+
+  validates :name, presence: true, length: { in: 3..20 }
+  validates :organization_id, presence: true
```

Comment:
what's the point of setting `optional: true` on an association when u validate presence?
also, never validate association by foreign_key, as it may be set to `-1` (for example) what would mean that record does not exists in DB